src/java/org/apache/cassandra/service/accord/AccordKeyspace.java

```diff
@@ -0,0 +1,843 @@
+        }
+
+        try
+        {
+            UntypedResultSet.Row row = result.one();
+            Preconditions.checkState(deserializeTimestampOrNull(row, "txn_id", TxnId::new).equals(txnId));
+            command.status.load(Status.values()[row.getInt("status")]);
+            command.homeKey.load(deserializeOrNull(row.getBlob("home_key"), AccordKey.serializer, row.getInt("home_key_version")));
+            command.progressKey.load(deserializeOrNull(row.getBlob("progress_key"), AccordKey.serializer, row.getInt("progress_key_version")));
+            command.isGloballyPersistent.load(row.getBoolean("is_globally_persistent"));
+            command.txn.load(deserializeOrNull(row.getBlob("txn"), CommandSerializers.txn, row.getInt("txn_version")));
+            command.executeAt.load(deserializeTimestampOrNull(row, "execute_at", Timestamp::new));
+            command.promised.load(deserializeTimestampOrNull(row, "promised_ballot", Ballot::new));
+            command.accepted.load(deserializeTimestampOrNull(row, "accepted_ballot", Ballot::new));
+            command.deps.load(deserializeWithVersionOr(row, "dependencies", "dependencies_version", CommandSerializers.deps, () -> Deps.NONE));
+            command.writes.load(deserializeWithVersionOr(row, "writes", "writes_version", CommandSerializers.writes, () -> null));
+            command.result.load(deserializeWithVersionOr(row, "result", "result_version", AccordData.serializer, () -> null));
+            command.waitingOnCommit.load(deserializeWaitingOn(row, "waiting_on_commit"));
+            command.blockingCommitOn.load(deserializeBlocking(row, "blocking_commit_on"));
+            command.waitingOnApply.load(deserializeWaitingOn(row, "waiting_on_apply"));
+            command.blockingApplyOn.load(deserializeBlocking(row, "blocking_apply_on"));
+            command.storedListeners.load(deserializeListeners(commandStore, row, "listeners"));
+        }
+        catch (IOException e)
+        {
+            logger.error("Exception loading AccordCommand " + command.txnId(), e);
+            throw new RuntimeException(e);
+        }
+        catch (Throwable t)
+        {
+            logger.error("Exception loading AccordCommand " + command.txnId(), t);
+            throw t;
+        }
+    }
+
+    private static void addSeriesMutations(AccordCommandsForKey.Series series,
+                                           PartitionUpdate.Builder partitionBuilder,
+                                           Row.Builder rowBuilder,
+                                           long timestampMicros,
+                                           int nowInSeconds)
+    {
+        if (!series.map.hasModifications())
+            return;
+
+        Row.Deletion deletion = series.map.hasDeletions() ?
+                                Row.Deletion.regular(new DeletionTime(timestampMicros, nowInSeconds)) :
+                                null;
+        ByteBuffer ordinalBytes = bytes(series.kind.ordinal());
```

Comment:
`ordinal` is brittle, can we store a value with the enum and use that?

Reply:
I'll address this and other equivalent places in my follow-up patch